openedx/core/djangoapps/content_tagging/rules.py

```diff
@@ -239,7 +239,7 @@ def can_view_object_tag_taxonomy(user: UserType, taxonomy: oel_tagging.Taxonomy)
 
     This rule is different from can_view_taxonomy because it checks if the taxonomy is enabled.
     """
-    return taxonomy.cast().enabled and can_view_taxonomy(user, taxonomy)
+    return not taxonomy or (taxonomy.cast().enabled and can_view_taxonomy(user, taxonomy))
```

Comment:
Hmm. I see now we have an issue here. This is not really the right fix, because it just disables taxonomy permission checking when you're not requesting a specific taxonomy.

The problem is that right now, the view code calls this with `taxonomy=None` rather than calling it several times, one for each taxonomy that is potentially in the result set. But we don't want to do the latter because it's going to be slow.

The _proper_ fix will be to [convert these specific permission checks to bridgekeeper](https://github.com/openedx/edx-platform/blob/master/lms/djangoapps/courseware/docs/decisions/0003-permissions-via-bridgekeeper.rst), and then use it to `.filter()` the queryset to enforce the permissions on a per-taxonomy basis without adding any additional queries. This is partially my fault as I suggested using django-rules here and didn't realize that bridgekeeper is now recommended instead (see also [OEP-66](https://docs.openedx.org/projects/openedx-proposals/en/latest/best-practices/oep-0066-bp-authorization.html#extending-permission-checks)). In retrospect we should have just used bridgekeeper from the beginning.

I'm also unsure if we have time left to do a big conversion to bridgekeeper, so: can we do a couple quick small PRs to change this specific check (`oel_tagging.view_objecttag`) to a bridgekeeper rule? Or is that going to take a lot of time?

Reply:
I don't know what it would take to convert some of our rules to bridgekeeper, sorry.

But we already already are filtering ObjectTags based on taxonomy.enabled & taxonomy orgs via the [ObjectTagTaxonomyOrgFilterBackend](https://github.com/openedx/edx-platform/blob/master/openedx/core/djangoapps/content_tagging/rest_api/v1/filters.py#L63).

Isn't that (plus my change) enough?

Reply:
Oh, I see. Great. I'm perhaps missing something else though... if we have that filter then why do we need this permissions check at all? Why not just change this rule to `return True  # perms are checked via ObjectTagTaxonomyOrgFilterBackend`

Reply:
The filter is used by the REST API, but these rules are also enforced in Django Admin.. don't know if that even matters.

But if we don't need this special `can_view_object_tag_taxonomy` rule, then we can remove it entirely, and do no special checking against the ObjectTag's taxonomy.

But something needs to be fixed, because as a non-superuser, I can't fetch a list of ObjectTags for an object.

Reply:
OK, let's go with what you have, but would you mind just adding a comment like `# Note: in the REST API, where we're dealing with multiple taxonomies at once, permissions are also enforced by ObjectTagTaxonomyOrgFilterBackend` ?
Then I think it makes sense.

Reply:
Done, thanks for this compromise, @bradenmacdonald !